packages/react-components/src/queries/useLatestAssetPropertyValues/useLatestAssetPropertyValues.ts

```diff
@@ -0,0 +1,184 @@
+ * (Note: can only be set once.)
+ * - defaults to 100ms
+ * @param batchSize configures the timeout window to accept
+ * (Note: can only be set once.)
+ * - defaults to 128
+ *
+ * @returns a list of QueryResults with GetAssetPropertyValueResponse data.
+ */
+export const useLatestAssetPropertyValues = ({
+  iotSiteWiseClient,
+  enabled = true,
+  requests = [],
+  refreshRate = DEFAULT_REFRESH_RATE,
+  batchSechedulerTimeout,
+  batchSize,
+}: UseLatestAssetPropertyValuesOptions) => {
+  /**
+   * A single enabled flag for all useLatestAssetPropertyValues
+   * with the same refreshRate. This will toggle the queries
+   * off / on in a setTimeout so that the executions all
+   * happen in the same task and the resulting refetchInterval
+   * executes within the batch timeout window.
+   */
+  const syncFlag = useSyncLatestAssetPropertyValueQueries({
+    enabled,
+    refreshRate,
+  });
+
+  const { getAssetPropertyValue, batchGetAssetPropertyValue } =
+    useIoTSiteWiseClient({ iotSiteWiseClient });
+
+  const queries = useMemo(() => {
+    return requests.map((request) => {
+      return {
+        enabled:
+          syncFlag &&
+          enabled &&
+          clientIsValid({
+            getAssetPropertyValue,
+            batchGetAssetPropertyValue,
+          }) &&
+          requestIsValid(request),
+        queryKey: new LatestAssetPropertyValueKeyFactory({
+          ...request,
+          refreshRate,
+        }).create(),
+        queryFn: createLatestAssetPropertyValueQueryFn(
+          {
+            getAssetPropertyValue,
+            batchGetAssetPropertyValue,
+          },
+          {
+            batchSechedulerTimeout,
+            batchSize,
+          }
+        ),
+        refetchInterval: refreshRate,
+      };
+    });
+  }, [
+    getAssetPropertyValue,
+    batchGetAssetPropertyValue,
+    enabled,
+    syncFlag,
+    requests,
+    refreshRate,
+    batchSechedulerTimeout,
+    batchSize,
+  ]);
+
+  return useQueries(
+    {
+      queries,
+    },
+    queryClient
+  );
+};
+
+export const createLatestAssetPropertyValueQueryFn = (
+  client?: QueryFnClient,
+  settings?: Pick<
+    UseLatestAssetPropertyValuesOptions,
+    'batchSechedulerTimeout' | 'batchSize'
+  >
+) => {
+  return async ({
+    queryKey: [{ assetId, propertyId, propertyAlias }],
+    signal,
+  }: QueryFunctionContext<
+    ReturnType<LatestAssetPropertyValueKeyFactory['create']>
+  >) => {
+    const { getAssetPropertyValue, batchGetAssetPropertyValue } = client ?? {};
+
+    invariant(
+      clientIsValid(client),
+      'Expected getAssetPropertyValue or batchGetAssetPropertyValue to be defined as required by the enabled flag.'
+    );
+
+    invariant(
+      requestIsValid({ assetId, propertyId, propertyAlias }),
+      'Expected assetId and propertyId to be defined or propertyAlias to be defined as required by the enabled flag.'
+    );
+
+    if (batchGetAssetPropertyValue) {
```

Comment:
So we inherently batch calls only if that requestFn exists? We may want to make this more explicit in the hook options.

Or something to consider: do we even want to use the non-batch function at all? Why not only support BatchGet*?

Reply:
Yes, we may not have access to batch calls, this is the case with for edge devices. The non batch apis ensures it still works in those contexts, but we of course prefer to use batch if available.

Reply:
Right edge! Good callout

Reply:
Added a note in the comments